doozerlib/rpmcfg.py

```diff
@@ -73,29 +73,7 @@ def __init__(self, runtime, data_obj, clone_source=True,
                 if self.public_upstream_branch:
                     self.private_fix = not util.is_commit_in_public_upstream(source_full_sha, self.public_upstream_branch, self.source_path)
 
-                # If this is a go project, parse the Godeps for points of interest
-                godeps_file = pathlib.Path(self.source_path, 'Godeps', 'Godeps.json')
-                if godeps_file.is_file():
-                    try:
-                        with open(str(godeps_file)) as f:
-                            godeps = json.load(f)
-                            # Reproduce https://github.com/openshift/origin/blob/6f457bc317f8ca8e514270714db6597ec1cb516c/hack/lib/build/version.sh#L82
-                            # Example of what we are after: https://github.com/openshift/origin/blob/6f457bc317f8ca8e514270714db6597ec1cb516c/Godeps/Godeps.json#L10-L15
-                            for dep in godeps.get('Deps', []):
-                                if dep.get('ImportPath', '') == 'k8s.io/kubernetes/pkg/api':
-                                    raw_kube_version = dep.get('Comment', '')  # e.g. v1.14.6-152-g117ba1f
-                                    # drop release information.
-                                    base_kube_version = raw_kube_version.split('-')[0]
-                                    kube_version_fields = base_kube_version.lstrip('v').split('.')  # v1.17.1 => [ '1', '17', '1' ]
-                                    # For historical consistency with tito's flow, we add +OS_GIT_COMMIT[:7] to the kube version
-                                    self.extra_os_git_vars['KUBE_GIT_VERSION'] = f"v{'.'.join(kube_version_fields)}+{source_full_sha[:7]}"
-                                    self.extra_os_git_vars['KUBE_GIT_COMMIT'] = dep.get('Rev', '')
-                                    self.extra_os_git_vars['KUBE_GIT_MAJOR'] = '0' if len(kube_version_fields) < 1 else kube_version_fields[0]
-                                    godep_kube_minor = '0' if len(kube_version_fields) < 2 else kube_version_fields[1]
-                                    self.extra_os_git_vars['KUBE_GIT_MINOR'] = f'{godep_kube_minor}+'  # For historical reasons, add a + since OCP patches its vendor kube.
-                    except:
-                        runtime.logger.error(f'Error parsing godeps {str(godeps_file)}')
-                        traceback.print_exc()
+                self.extra_os_git_vars.update(self.metadata.extract_kube_env_vars())
```

Comment:
The failure caused by the transition from godep to gomod was only discovered by manual inspection of the build artifacts. If kube env vars cannot  be populated for some reason, would it make sense for that condition to be logged or raised as an error to allow this class of failure to be detected automatically in the future?

Reply:
I hate to have any image specific code in the tool, but this seems important enough to make an exception. I've added a check in the next commit. If the information is not available for hyperkube, an exception will be thrown.
```
OSError: Unable to determine KUBE vars for openshift-enterprise-hyperkube
2020-08-06 19:07:53,234 ERROR [containers/openshift-enterprise-hyperkube] Unable to determine KUBE vars for openshift-enterprise-hyperkube
```